### ISSUE TITLE
Failed to install scylla on Suse Enterprise 15 for _yaml.so dependency

### ISSUE THREAD

**Issue by amoskong** [scylladb/scylla#9045]
*Installation details*
Scylla version (or git commit hash): 
- scylla-4.5.rc4-0.20210707.9e55d9bd0.x86_64.rpm
- scylla-tools-core-4.5.rc4-0.20210707.9e55d9bd0.noarch.rpm
Cluster size: 1
OS (RHEL/CentOS/Ubuntu/AWS AMI): Suse Enterprise 15

### Description

Takuya already added support of Suse in commit https://github.com/scylladb/scylla/commit/bcd41cb32d68d57140c310763bffa6b1befd301a
- (Merge 'Support installing our rpm to SLES' from Takuya ASADA)

### Failed to install scylla on suse 15 for yaml.so dependency.

```
# wget http://downloads.scylladb.com.s3.amazonaws.com/rpm/centos/scylla-4.5.repo
# sudo cp scylla-4.5.repo /etc/zypp/repos.d

# sudo zypper install scylla
Building repository 'Scylla for Centos - x86_64' cache ......................................................................................[done]
Building repository 'Scylla for centos' cache ...............................................................................................[done]
Loading repository data...
Reading installed packages...
Resolving package dependencies...

Problem: nothing provides '/usr/lib64/python2.7/site-packages/_yaml.so' needed by the to be installed scylla-tools-core-4.5.rc4-0.20210707.9e55d9bd0.noarch
 Solution 1: do not install scylla-4.5.rc4-0.20210707.9e55d9bd0.x86_64
 Solution 2: break scylla-tools-core-4.5.rc4-0.20210707.9e55d9bd0.noarch by ignoring some of its dependencies

Choose from above solutions by number or cancel [1/2/c/d/?] (c): 1
Resolving dependencies...
Resolving package dependencies...
Nothing to do.
```

```
scylla-test@amos-suse:/etc/zypp/repos.d> sudo zypper search python2-PyYAML
Loading repository data...
Reading installed packages...

S | Name           | Summary                            | Type
--+----------------+------------------------------------+--------
  | python2-PyYAML | YAML parser and emitter for Python | package
```

### The problem isn't solved after installed python2-PyYAML

```
scylla-test@amos-suse:/etc/zypp/repos.d> sudo zypper install python2-PyYAML
Refreshing service 'Basesystem_Module_x86_64'.
Refreshing service 'Containers_Module_x86_64'.
Refreshing service 'Desktop_Applications_Module_x86_64'.
Refreshing service 'Development_Tools_Module_x86_64'.
Refreshing service 'Legacy_Module_x86_64'.
Refreshing service 'Public_Cloud_Module_x86_64'.
Refreshing service 'Python_2_Module_x86_64'.
Refreshing service 'SUSE_Cloud_Application_Platform_Tools_Module_x86_64'.
Refreshing service 'SUSE_Linux_Enterprise_Server_x86_64'.
Refreshing service 'Server_Applications_Module_x86_64'.
Refreshing service 'Web_and_Scripting_Module_x86_64'.
Loading repository data...
Reading installed packages...
Resolving package dependencies...

The following 4 NEW packages are going to be installed:
  libpython2_7-1_0 python2-PyYAML python-base shared-python-startup

4 new packages to install.
Overall download size: 6.7 MiB. Already cached: 0 B. After the operation, additional 26.2 MiB will be used.
Continue? [y/n/v/...? shows all options] (y): y
Retrieving package libpython2_7-1_0-2.7.18-7.55.1.x86_64                                                      (1/4), 834.6 KiB (  2.2 MiB unpacked)
Retrieving: libpython2_7-1_0-2.7.18-7.55.1.x86_64.rpm .......................................................................................[done]
Retrieving package shared-python-startup-0.1-1.5.1.noarch                                                     (2/4),  13.1 KiB ( 13.5 KiB unpacked)
Retrieving: shared-python-startup-0.1-1.5.1.noarch.rpm ......................................................................................[done]
Retrieving package python-base-2.7.18-7.55.1.x86_64                                                           (3/4),   5.8 MiB ( 23.5 MiB unpacked)
Retrieving: python-base-2.7.18-7.55.1.x86_64.rpm ............................................................................................[done]
Retrieving package python2-PyYAML-5.4.1-1.1.x86_64                                                            (4/4), 104.0 KiB (481.1 KiB unpacked)
Retrieving: python2-PyYAML-5.4.1-1.1.x86_64.rpm .............................................................................................[done]

Checking for file conflicts: ................................................................................................................[done]
(1/4) Installing: libpython2_7-1_0-2.7.18-7.55.1.x86_64 .....................................................................................[done]
(2/4) Installing: shared-python-startup-0.1-1.5.1.noarch ....................................................................................[done]
(3/4) Installing: python-base-2.7.18-7.55.1.x86_64 ..........................................................................................[done]
(4/4) Installing: python2-PyYAML-5.4.1-1.1.x86_64 ...........................................................................................[done]
scylla-test@amos-suse:/etc/zypp/repos.d> ls /usr/lib64/python2.7/site-packages/_yaml.so
ls: cannot access '/usr/lib64/python2.7/site-packages/_yaml.so': No such file or directory
scylla-test@amos-suse:/etc/zypp/repos.d> ls /usr/lib64/python2.7/site-packages
_local.pth  PyYAML-5.4.1-py2.7.egg-info  README  _yaml  yaml
```

### System info

```
scylla-test@amos-suse:~> uname -a
Linux amos-suse 5.3.18-59.10-default #1 SMP Fri Jun 25 12:36:56 UTC 2021 (6856d31) x86_64 x86_64 x86_64 GNU/Linux


scylla-test@amos-suse:~> cat /etc/os-release
NAME="SLES"
VERSION="15-SP3"
VERSION_ID="15.3"
PRETTY_NAME="SUSE Linux Enterprise Server 15 SP3"
ID="sles"
ID_LIKE="suse"
ANSI_COLOR="0;32"
CPE_NAME="cpe:/o:suse:sles:15:sp3"
DOCUMENTATION_URL="https://documentation.suse.com/"
```

/Cc @roydahan

**Comment by amoskong** [comment]
In https://rpmfind.net/linux/RPM/opensuse/15.1/x86_64/python2-PyYAML-3.13-lp151.1.1.x86_64.html

I saw the `/usr/lib64/python2.7/site-packages/_yaml.so` is in the Files list.

But after installed python2-PyYAML by zypper, I can't fine the _yaml.so file.

scylla-test@amos-suse:/etc/zypp/repos.d> sudo find / |grep yaml.*.so
/usr/lib64/libyaml-0.so.2
/usr/lib64/libyaml-cpp.so.0.6
/usr/lib64/libyaml-0.so.2.0.5
/usr/lib64/libyaml-cpp.so.0.6.0
/usr/lib64/python2.7/site-packages/yaml/resolver.py
/usr/lib64/python2.7/site-packages/yaml/resolver.pyc
/usr/lib64/python2.7/site-packages/yaml/resolver.pyo

I can successfully import the yaml module.

```
scylla-test@amos-suse:/etc/zypp/repos.d> python
Python 2.7.18 (default, Mar 04 2021, 23:25:57) [GCC] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import yaml
>>> 
```

**Comment by amoskong** [comment]
@syuu1228 

If we choose solution 2 in scylla install to skip the yaml.so dependency,
- Solution 2: break scylla-tools-core-4.5.rc4-0.20210707.9e55d9bd0.noarch by

Scylla will be installed successfully, and scylla-server can be started, cqlsh/nodetool/cassandra-stress works well.

#### Detail logs:

[scylla-install-setup-log-on-suse-15.txt](https://github.com/scylladb/scylla/files/6820889/scylla-install-setup-log-on-suse-15.txt)

**Comment by amoskong** [comment]
python2-PyYAML is no longer available in default repo of latest SUSE Linux Enterprise Server 15 SP3.
It's available on last month (July 15).

I have to download the python2-PyYAML package from rpmfind.net

- https://rpmfind.net/linux/RPM/opensuse/15.1/x86_64/python2-PyYAML-3.13-lp151.1.1.x86_64.html
- https://rpmfind.net/linux/RPM/opensuse/15.3/x86_64/python2-PyYAML-5.4.1-1.1.x86_64.html  (latest one)

It's not grace to add this in our install guide.

**Comment by syuu1228** [comment]
@amoskong I guess it related with deprecating Python2 on SP3.
https://www.suse.com/releasenotes/x86_64/SUSE-SLES/15-SP3/

_5.5.11 Python 2 is deprecated_
_The python executable is only provided via the Python 2 module, not via the default repositories._

On GCP image, Python2 module is enabled by default, but probably on other image it disabled.

I think following command enables the module:
```
sudo zypper mr -e Python_2_Module_x86_64:SLE-Module-Python2-15-SP3-Pool
sudo zypper mr -e Python_2_Module_x86_64:SLE-Module-Python2-15-SP3-Updates
```

**Comment by syuu1228** [comment]
@amoskong I found another problem after we fixed this issue: #9168

**Comment by avikivity** [comment]
Minor issue (rarely used distribution) so not backporting unless there is a user request.
